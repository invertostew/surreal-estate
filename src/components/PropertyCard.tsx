import React from "react";

import "../styles/PropertyCard.css";

interface PropertyCardProps {
  title: string;
  type: string;
  city: string;
  bathrooms: number;
  bedrooms: number;
  price: number;
  email: string;
}

const PropertyCard: React.FC<PropertyCardProps> = ({
  title,
  type,
  city,
  bathrooms,
  bedrooms,
  price,
  email,
}): JSX.Element => {
  return (
    <article className="property-card">
      <p className="property-card__title">Title: {title}</p>
      <p className="property-card__type">Type: {type}</p>
      <p className="property-card__city">City: {city}</p>
      <p className="property-card__bathrooms">Bathrooms: {bathrooms}</p>
      <p className="property-card__bedrooms">Bedrooms: {bedrooms}</p>
      <p className="property-card__price">Price: {price}</p>
      <p className="property-card__email">Email: {email}</p>
    </article>
  );
};

export default PropertyCard;
