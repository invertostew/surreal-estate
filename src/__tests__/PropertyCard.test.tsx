import React from "react";
import { render, screen } from "@testing-library/react";

import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Test Title",
    type: "Test Type",
    city: "Test City",
    bathrooms: 2,
    bedrooms: 3,
    price: 234995,
    email: "person@email.com",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        city={validProps.city}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        email={validProps.email}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct values for props", () => {
    render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        city={validProps.city}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        email={validProps.email}
      />
    );

    const titleEl = screen.getByText(/test title/i);
    const typeEl = screen.getByText(/test type/i);
    const cityEl = screen.getByText(/test city/i);
    const bathroomsEl = screen.getByText(/bathrooms: 2/i);
    const bedroomsEl = screen.getByText(/bedrooms: 3/i);
    const priceEl = screen.getByText(/234995/i);
    const emailEl = screen.getByText(/person@email.com/i);

    expect(titleEl).toBeInTheDocument();
    expect(titleEl).toHaveClass("property-card__title");

    expect(typeEl).toBeInTheDocument();
    expect(typeEl).toHaveClass("property-card__type");

    expect(cityEl).toBeInTheDocument();
    expect(cityEl).toHaveClass("property-card__city");

    expect(bathroomsEl).toBeInTheDocument();
    expect(bathroomsEl).toHaveClass("property-card__bathrooms");

    expect(bedroomsEl).toBeInTheDocument();
    expect(bedroomsEl).toHaveClass("property-card__bedrooms");

    expect(priceEl).toBeInTheDocument();
    expect(priceEl).toHaveClass("property-card__price");

    expect(emailEl).toBeInTheDocument();
    expect(emailEl).toHaveClass("property-card__email");
  });
});
