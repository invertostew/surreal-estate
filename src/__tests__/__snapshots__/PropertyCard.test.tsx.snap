// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <article
    class="property-card"
  >
    <p
      class="property-card__title"
    >
      Title: Test Title
    </p>
    <p
      class="property-card__type"
    >
      Type: Test Type
    </p>
    <p
      class="property-card__city"
    >
      City: Test City
    </p>
    <p
      class="property-card__bathrooms"
    >
      Bathrooms: 2
    </p>
    <p
      class="property-card__bedrooms"
    >
      Bedrooms: 3
    </p>
    <p
      class="property-card__price"
    >
      Price: 234995
    </p>
    <p
      class="property-card__email"
    >
      Email: person@email.com
    </p>
  </article>
</DocumentFragment>
`;
